### ISSUE TITLE
VdomAttr[Any] cannot be assigned a double

### ISSUE THREAD

**Issue by ramnivas** [japgolly/scalajs-react#388]
The following code fails:

```scala
input(..., step := 0.05)
```

with the error:
```
Error:(34, 18) ambiguous implicit values:
 both method vdomAttrVtKey in trait ImplicitsForVdomAttr1 of type [A](implicit k: A => japgolly.scalajs.react.raw.Key)japgolly.scalajs.react.vdom.Attr.ValueType[A,japgolly.scalajs.react.vdom.Attr.Key]
 and method vdomAttrVtJsAny in trait ImplicitsForVdomAttr of type [A](implicit f: A => scala.scalajs.js.Any)japgolly.scalajs.react.vdom.Attr.ValueType[A,Any]
 match expected type japgolly.scalajs.react.vdom.Attr.ValueType[Double,Any]
        step := 0.05
```

Looking at https://github.com/japgolly/scalajs-react/blob/cb75721e3bbd0033ad63d380bcaddc96fbe906e3/test/src/test/scala/japgolly/scalajs/react/core/vdom/ReactAttrTest.scala#L31 the test for double is missing (wouldn't have compiled anyway).

Is this intended to work? I have the workaround of specifying "0.05", but it seems 0.05 should have also worked.

Somewhat aside: I think all metered values (max, min, step) should have be of `VdomAttr[Double]`. I can make a PR to that effect, but wanted to know if I missed something.

**Comment by japgolly** [comment]
Good catch! That's the first bug I've been in 1.0 :tada:
I think that's an oversight on my part. I thought it would be covered in `vdom/PrefixedTest.scala` but I see it's only on the `key attribute: https://github.com/japgolly/scalajs-react/blob/cb75721e3bbd0033ad63d380bcaddc96fbe906e3/test/src/test/scala/japgolly/scalajs/react/core/vdom/PrefixedTest.scala#L265

About `step` being a `VdomAttr[Double]` instead of `VdomAttr[Any]`, I had a quick look online and apparently `"any"` is valid too.

**Comment by japgolly** [comment]
Oh and FYI, not all vdom attributes have precise types. The most common ones do but many are just defined as `VdomAttr[Any]` because it's too time-consuming for me to do everything. If you come across attributes that you know should have a more precise type please feel free to tweak in a PR (and spread the word :grinning:). It's going to have to be an incremental effort.

**Comment by ramnivas** [comment]
First bug in 1.0 took this long... that is quite a feat (yours!) and indication of the quality of code you have produced!

I will definitely look for the possibilities for more precise types in this context and produce PRs. I have a relatively large code that I am mostly done and so far this is all I found (in this context).

**Comment by japgolly** [comment]
Thanks @ramnivas! :bowing_man:

**Comment by japgolly** [comment]
Fixed via b9fa89c800d9346ea4a0c41962ea9bda91db1e9e.
Closing. Feel free to reopen if the problem persists in v1.1.0
